validator/server_api/validation_api.go

```diff
@@ -122,6 +126,27 @@ func (a *ExecServerAPI) removeOldRuns(ctx context.Context) time.Duration {
 func (a *ExecServerAPI) Start(ctx_in context.Context) {
 	a.StopWaiter.Start(ctx_in, a)
 	a.CallIteratively(a.removeOldRuns)
+	if a.consumer != nil {
+		a.consumer.Start(ctx_in)
+		a.StopWaiter.CallIteratively(func(ctx context.Context) time.Duration {
+			msg, err := a.consumer.Consume(ctx)
+			if err != nil {
+				log.Error("Consuming request", "error", err)
+				// TODO: consider having this as config param.
+				return time.Second
+			}
+			hashes, err := a.GetLeavesWithStepSize(ctx, msg.Value.ExecutionID, msg.Value.FromBatch, msg.Value.MachineStartIndex, msg.Value.StepSize, msg.Value.NumDesiredLeaves)
+			if err != nil {
+				// TODO: log other fields as well.
+				log.Error("Error getting leaves with stepsize", "error", err)
+			}
+			if err := a.consumer.SetResult(ctx, msg.ID, hashes); err != nil {
+				log.Error("Error setting result", "error", err)
+			}
+			return time.Second
```

Comment:
waiting a second makes sense in case of an error, but if no error you should return 0 so it would go back to work immediately